Tests/test_SearchIO_hmmer3_text.py

```diff
@@ -1654,6 +1654,13 @@ def test_30_hmmscan_010(self):
         self.assertEqual(0, hit.domain_obs_num)
         self.assertEqual(0, len(hit))
 
+    def test_30_hmmscan_011(self):
+        """Parsing hmmscan 3.0 (text_30_hmmscan_011)."""
+        hmmer_file = get_file("text_30_hmmscan_011.out")
+        # Test getting program name when preamble contains full path and whitespace
+        for qresult in parse(hmmer_file, FMT):
+            assert qresult.program == "hmmscan"
+
```

Comment:
This is a good start :+1:. One thing I might suggest here is to either add a little more detail to the assertion, for when the test fails.

Something like:
```
...
    assert qresult.program == "hmmscan", f"Expected 'hmmscan' in item index {idx}, found {qresult.program!r}"
...
```


Furthermore, let's add a little bit more for some light sanity checks. Would you mind adding:

* A test to make sure we have the expected number of `qresult`?
* For each qresult, a test to make sure have parsed its contents correctly. I'm thinking just testing for the number of parsed hits is enough.
* Optionally, also a test to make sure all HSPs are parsed correctly. Nothing too involved, something like the `score` of the last HSP should be enough.

Reply:
done! let me know what you think